pyxrf/xanes_maps/xanes_maps_api.py

```diff
@@ -2934,6 +3036,59 @@ def _save_xanes_maps_to_tiff(*, wd, eline_data_aligned, eline_selected,
             print(f"\nR-factors for XANES maps are saved to file '{fln_xanes_rfactor}'.", file=f_log)
 
 
+def _save_spectrum_as_csv(*, fln, wd=None, msg_info=None, energy=None, spectrum=None):
+    """
+    Saves spectrum as CSV file.
+
+    Parameters
+    ----------
+
+    fln: str
+        name of the file to save spectrum
+
+    wd: str
+        working directory where the file ``file`` is to be placed
+
+    msg_info: str
+        A string that will be placed at the beginning of the CSV file before the file header.
+        Each line in the string must start with # (to separate the comment).
+
+    energy: ndarray(float)
+        The array of energy values in keV
+
+    spectrum: ndarray(float)
+        The array of spectrum values. Must have the same size as 'energy'.
+    """
+
+    # Full file path
+    file_path = os.path.join(wd, fln) if wd else fln
+
+    msg = None
+    if energy is None:
+        msg = "The array 'energy' is None"
+    elif spectrum is None:
+        msg = "The array 'spectrum' is None"
+    elif len(energy) != len(spectrum):
+        msg = "The arrays 'energy' and 'spectrum' have different size"
```

Comment:
It feels like reinvention of a wheel. As I mentioned in our conversation yesterday, pandas [dataframe's `to_csv()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html) method can be uses for more efficient exporting and to avoid potential issues which a huge community has already solved.

As you would pass a numpy array with 2 columns, each column must have the same size, or otherwise it will fail on the array creation. As a result, this extra check may not be needed.

Unfortunately, pandas' `to_csv()` it does not support the comments field, but it can be done with NumPy's [`savetext()`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.savetxt.html).